### PR TITLE
Remove deprecated props from examples

### PR DESCRIPTION
**Pull request by raiju**
#### Checklist

- [X] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [X] Include tests (Not Applicable)
- [X] Update documentation (Not Applicable)

#### Changes proposed in this pull request:

Remove usage of deprecated props in examples/docs

### CODE REVIEW

File: packages/table/preview/features.tsx [hunk 1]
```diff
@@ -183,7 +183,10 @@ class EditableTable extends React.Component<{}, IEditableTableState> {
             <Column key={index} renderCell={this.renderCell} renderColumnHeader={this.renderColumnHeader} />
         ));
         return (
-            <Table numRows={7} selectionModes={SelectionModes.COLUMNS_AND_CELLS} enableFocus={true}>
+            <Table numRows={7}
+                   selectionModes={SelectionModes.COLUMNS_AND_CELLS}
+                   enableFocus={true}
+                   useInteractionBar={true}>
```
Comment (cmslewis) [inline packages/table/preview/features.tsx]: Hmm, is `prettier` fixing/complaining about this formatting? Should be. Change to:

```tsx
<Table
    numRows={7}
    selectionModes={SelectionModes.COLUMNS_AND_CELLS}
    enableFocus={true}
    useInteractionBar={true}
>
```

File: packages/table/test/columnHeaderCellTests.tsx [hunk 2]
```diff
@@ -75,9 +75,10 @@ describe("<ColumnHeaderCell>", () => {
         it("renders custom menu items", () => {
             const menuClickSpy = sinon.spy();
             const menu = getMenuComponent(menuClickSpy);
+            const menuFn = () => menu;
```
Comment (cmslewis) [inline packages/table/test/columnHeaderCellTests.tsx]: nit: Just call it `renderMenu`.

File: packages/table/preview/features.tsx (hunk 1 above)
Comment (cmslewis) [reply packages/table/preview/features.tsx]: Actually the build is failing due to several lint issues.

**Comment by cmslewis** [comment]
Failures:
image

**Comment by giladgray** [comment]
@raiju will you have a change to fix the lint errors this week? should be as easy as `gulp tslint-fix`.

**Comment by raiju** [comment]
@cmslewis @giladgray My bad, forgot to check my notifications. Those issues should be fixed.

**Comment by raiju** [comment]
<h3>Fix linting</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/raiju/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/raiju/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/raiju/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/raiju/blueprint/5/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by raiju** [comment]
<h3>Address nit</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/raiju/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/raiju/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/raiju/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/raiju/blueprint/6/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by cmslewis** [comment]
Thanks @raiju!

**Pull request merged**